File: experiments/tools/contentprovider/compose/activity_main.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun activity_main() {
  Box(modifier = Modifier.fillMaxWidth().fillMaxHeight().padding(bottom = 5.0.dp)) {
    Box(modifier = Modifier.align(Alignment.TopStart)) {
      val spinnerOptions_spinner_content_providers = listOf("Select", "Option 1", "Option 2", "Option 3")
      var spinnerExpanded_spinner_content_providers by remember { mutableStateOf(false) }
      var spinnerSelected_spinner_content_providers by remember { mutableStateOf(spinnerOptions_spinner_content_providers.first()) }
      Box(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
        OutlinedButton(onClick = { spinnerExpanded_spinner_content_providers = true }, modifier = Modifier.fillMaxWidth()) {
          Text(text = spinnerSelected_spinner_content_providers)
        }
        DropdownMenu(expanded = spinnerExpanded_spinner_content_providers, onDismissRequest = { spinnerExpanded_spinner_content_providers = false }) {
          spinnerOptions_spinner_content_providers.forEach { option ->
            DropdownMenuItem(
              text = { Text(option) },
              onClick = {
                spinnerSelected_spinner_content_providers = option
                spinnerExpanded_spinner_content_providers = false
              }
            )
          }
        }
      }
    }
    Box(modifier = Modifier.align(Alignment.TopStart)) {
      Box(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
        Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
          Text(text = "Show Column Types")
        }
        Box(modifier = Modifier.align(Alignment.TopStart)) {
          Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
            Text(text = "Uncheck All")
          }
        }
        Box(modifier = Modifier.align(Alignment.TopEnd)) {
          Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
            Text(text = "Check All")
          }
        }
      }
    }
    Box(modifier = Modifier.align(Alignment.TopStart)) {
      Text(text = "Columns:", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(end = 4.0.dp))
    }
    Box(modifier = Modifier.align(Alignment.TopEnd)) {
      Text(text = "-", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(bottom = 5.0.dp))
    }
    Box(modifier = Modifier.align(Alignment.TopStart)) {
      val listItems_list_view_columns = (1..10).map { it }
      LazyColumn(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
        items(listItems_list_view_columns) { idx ->
          Column(modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp)) {
            Text(text = "Item $idx")
            Text(text = "Sub Item $idx")
          }
        }
      }
    }
    Box(modifier = Modifier.align(Alignment.BottomStart)) {
      Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(top = 5.0.dp), horizontalAlignment = Alignment.CenterHorizontally) {
        Row(modifier = Modifier.fillMaxWidth().wrapContentHeight(), horizontalArrangement = Arrangement.Center, verticalAlignment = Alignment.CenterVertically) {
          Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
            Text(text = "Query")
          }
          Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
            Text(text = "Query With Filter")
          }
        }
        Row(modifier = Modifier.fillMaxWidth().wrapContentHeight(), horizontalArrangement = Arrangement.Center, verticalAlignment = Alignment.CenterVertically) {
          Text(text = "Rows:", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(end = 4.0.dp))
          Text(text = "-", modifier = Modifier.wrapContentWidth().wrapContentHeight())
        }
      }
    }
  }
}


@Preview(showBackground = true)
@Composable
fun activity_main_preview() {
    MyApplicationTheme {
        activity_main()
    }
}

File: experiments/tools/contentprovider/swiftui/Activity_main.swift
import SwiftUI

struct Activity_main: View {
  var body: some View {
    ZStack {
      Group {
      Picker("", selection: .constant(0)) {
        Text("Select").tag(0)
        Text("Option 1").tag(1)
        Text("Option 2").tag(2)
        Text("Option 3").tag(3)
      }
      .pickerStyle(.menu)
      .frame(maxWidth: .infinity)
      }
      .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
      Group {
      ZStack {
        Button(action: { }) {
          Text("Show Column Types").frame(maxWidth: .infinity, alignment: .leading)
        }
        .disabled(false)
        Group {
        Button(action: { }) {
          Text("Uncheck All").frame(maxWidth: .infinity, alignment: .leading)
        }
        .disabled(false)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
        Group {
        Button(action: { }) {
          Text("Check All").frame(maxWidth: .infinity, alignment: .leading)
        }
        .disabled(false)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topTrailing)
      }
      .frame(maxWidth: .infinity)
      }
      .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
      Text("Columns:").padding(.trailing, 4.0).frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
      Text("-").padding(.bottom, 5.0).frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topTrailing)
      Group {
      ScrollView {
        LazyVStack(alignment: .leading, spacing: 4) {
          ForEach(1...10, id: \.self) { idx in
            VStack(alignment: .leading, spacing: 2) {
              Text("Item \(idx)")
              Text("Sub Item \(idx)")
            }
            Divider()
          }
        }
      }
      .frame(maxWidth: .infinity, maxHeight: .infinity)
      }
      .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
      Group {
      VStack(alignment: .center, spacing: 0) {
        HStack(alignment: .center, spacing: 0) {
          Button(action: { }) {
            Text("Query").frame(maxWidth: .infinity, alignment: .leading)
          }
          .disabled(false)
          Button(action: { }) {
            Text("Query With Filter").frame(maxWidth: .infinity, alignment: .leading)
          }
          .disabled(false)
        }
        .frame(maxWidth: .infinity)
        HStack(alignment: .center, spacing: 0) {
          Text("Rows:").padding(.trailing, 4.0)
          Text("-")
        }
        .frame(maxWidth: .infinity)
      }
      .frame(maxWidth: .infinity)
      .padding(.top, 5.0)
      }
      .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .bottomLeading)
    }
    .frame(maxWidth: .infinity, maxHeight: .infinity)
    .padding(.bottom, 5.0)
  }
}

struct Activity_main_Previews: PreviewProvider {
  static var previews: some View {
    Activity_main()
  }
}
